stock_component/models/stock_move.py

```diff
@@ -25,3 +31,33 @@ def _has_components(self):
 
     def generate_component_moves(self):
         self.mapped("move_line_ids.lot_id").pull_components()
+
+    def _is_pulling_components(self):
+        return self._context.get(PULLING_COMPONENTS, False)
+
+    def _check_no_component_moved(self):
+        serials = self.mapped("move_line_ids.lot_id")
+        component_serials = serials.filtered(lambda s: s.is_component)
+
+        if component_serials:
+            message = _(
+                "You are attempting to move products that are components "
+                "of an equipment."
+            )
+            details = "\n".join(
+                [
+                    self._get_component_serial_not_movable_message(s)
+                    for s in component_serials
+                ]
+            )
```

Comment:
pas besoin de mettre ça dans une liste. Tu peux directement mettre l'iterateur.

```suggestion
            details = "\n".join(
                self._get_component_serial_not_movable_message(s)
                for s in component_serials
            )
```
PS: pas sur de l'indentation ;(

Reply:
merci